advanced_survey/views.py

```diff
@@ -93,7 +124,15 @@ def save_survey(request): # pylint: disable=R0912,R0915
                             if isinstance(response, str):
                                 raise RuntimeError(response)
                         db_question.options = json.dumps(options)
+                    
+                    
+
+                    db_question.is_visible = parse_external_reference(question["is_visible"], id_lookup, False)
+                    db_question.is_required = parse_external_reference(question["is_required"], id_lookup, True)
+
+                    temp_id = question["id"]
                     db_question.save()
+                    id_lookup[temp_id] = db_question.pk
```

Comment:
Is there a reason you used a temp variable?

Reply:
Oh, my mistake. Got my wires crossed and thought that calling save would change it. Will fix